### PR TITLE
if NiftiImageData3D, run check

### PR DESCRIPTION
**Pull request by rijobro**
Fixes https://github.com/SyneRBI/SIRF/issues/816.

@KrisThielemans -- Haven't compiled/tested but hopefully the logic should be clear here. The problem was that the python constructors for `NiftiImageData` and `NiftiImageData3D` but used the C++ `NiftiImageData` constructor, so there were no checks taking place.

### CODE REVIEW

File: src/Registration/cReg/cReg.cpp [hunk 1]
```diff
@@ -447,13 +447,16 @@ void* cReg_NiftiImageData_get_inner_product(const void* im1_ptr, const void* im2
...
+void* cReg_NiftiImageData_from_SIRFImageData(void* ptr, const int is_3D)
 {
-	try {
+    try {
         ImageData& sirf_im = objectFromHandle<ImageData>(ptr);
-        std::shared_ptr<NiftiImageData<float> >
-            sptr(new NiftiImageData<float>(sirf_im));
-        return newObjectHandle(sptr);
+
+        // Use an int as if bool to optionally run 3D checks
```
Comment (KrisThielemans) [inline src/Registration/cReg/cReg.cpp:455]: cryptic!

Comment (KrisThielemans) [reply src/Registration/cReg/cReg.cpp:455]: I maybe I understand, maybe we can't wrap `bool`? fine. I'd put this in the doxygen documentation  of the wrapper then

**Review by KrisThielemans — CHANGES_REQUESTED** [review]

File: src/Registration/cReg/cReg.cpp [hunk 2]
```diff
@@ -447,13 +447,16 @@ void* cReg_NiftiImageData_get_inner_product(const void* im1_ptr, const void* im2
...
+        if (is_3D == 1)
```
Comment (KrisThielemans) [inline src/Registration/cReg/cReg.cpp:456]:
```suggestion
        if (is_3D)
```

**Comment by KrisThielemans** [comment]
looks good. I'll try this later I guess and merge then. thx!

File: src/Registration/cReg/cReg.cpp (hunk 1 above)
Comment (rijobro) [reply src/Registration/cReg/cReg.cpp]: Yeah, as far as I know we can pass `int` without any wrapping, but I don't think the same is true for booleans.

File: src/Registration/cReg/cReg.cpp (hunk 2 above)
Comment (rijobro) [reply src/Registration/cReg/cReg.cpp]: Can be, I put it in to be explicit as I was using `int` instead of `bool`. Feel free to commit the suggestion.

**Review by KrisThielemans — APPROVED** [review]

**Pull request merged**